src/main/java/ch/njol/skript/expressions/ExprCoordinate.java

```diff
@@ -59,8 +61,8 @@ public boolean init(final Expression<?>[] exprs, final int matchedPattern, final
 	}
 	
 	@Override
-	public Number convert(final Location l) {
-		return axis == 0 ? l.getX() : axis == 1 ? l.getY() : l.getZ();
+	public Number convert(final Object o) {
+		return (int) (o instanceof Location ? axis == 0 ? ((Location) o).getX() : axis == 1 ? ((Location) o).getY() : ((Location) o).getZ() : axis == 0 ? ((Chunk) o).getX() : axis == 1 ? 0 : ((Chunk) o).getZ());
```

Comment:
This better be splitted into if-else for readability

Reply:
the weird thing is that apparently my changes didn't get posted :D because what you're requesting here is very similar to my final part, without the final and some little things :D

Reply:
If you're editing from you mobile still I recommend you to use a PC/laptop instead

Reply:
Still unresolved

Reply:
i will look on it, when ill be home